test/parser_test.dart

```diff
@@ -49,14 +46,65 @@ void main() {
   });
 }
 
+const ceil = 100000;
+
 const _psr = const Parser();
 
+late Random rand;
+
 class ParserTest {
-  void parseOne_commentQuote_terminalChar2() {
-    var fail =
-        _psr.parseOne('fruit = banana # I\'m a comment with a final "quote"');
-    expect(
-        fail['fruit'], equals('banana # I\'m a comment with a final "quote"'));
+  void interpolate() {
+    var out = _psr.interpolate(r'a$foo$baz', {'foo': 'bar', 'baz': 'qux'});
+    expect(out, equals('abarqux'));
+  }
+
+  void interpolate_curlies() {
+    var r = rand.nextInt(ceil); // avoid runtime collision with real env vars
+    var out = _psr.interpolate('optional_\${foo_$r}', {'foo_$r': 'curlies'});
+    expect(out, equals('optional_curlies'));
+  }
+
+  void interpolate_fallback() {
+    var out = _psr.interpolate('a\$HOME', {});
+    expect(out, equals('a${Platform.environment['HOME']}'));
+  }
+
+  void interpolate_missing() {
+    var r = rand.nextInt(ceil); // avoid runtime collision with real env vars
+    var out = _psr.interpolate('a\$jinx_$r', {});
+    expect(out, equals('a'));
+  }
+
+  void interpolate_missing2() {
+    var r = rand.nextInt(ceil); // avoid runtime collision with real env vars
+    var out = _psr.interpolate('a\$foo_$r\$baz_$r', {'foo_$r': ''});
+    expect(out, equals('a'));
+  }
+
+  void parse_dup() {
+    var out = _psr.parse(['foo=bar', 'foo=baz']);
+    expect(out, equals({'foo': 'bar'}));
```

Comment:
I think on duplicate we should return the latest value.
I confirmed this behaviour with DotEnv in PHP. Not sure how this is handled in others